usersession/userd/launcher.go

```diff
@@ -183,12 +189,190 @@ func (s *Launcher) OpenURL(addr string, sender dbus.Sender) *dbus.Error {
 	return nil
 }
 
+// OpenDesktopEntry implements the 'OpenDesktopEntry' method of the 'io.snapcraft.DesktopLauncher'
+// DBus interface. The desktopFileID is described here:
+// https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#desktop-file-id
+func (s *Launcher) OpenDesktopEntry(desktopFileID string, sender dbus.Sender) *dbus.Error {
+	desktopFile, err := desktopFileIDToFilename(osutil.RegularFileExists, desktopFileID)
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	err = verifyDesktopFileLocation(desktopFile)
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	exec_command, err := readExecCommandFromDesktopFile(desktopFile)
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	args, err := parseExecCommand(exec_command)
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+  args = append([]string{"systemd-run", "--user", "--"}, args...)
```

Comment:
It's not immediately obvious that inheriting the environment of userd is going to be particularly valuable here: I wouldn't expect userd to have anything more useful in its environment than systemd itself (in fact, I'd expect them to be nearly identical).

Also, if we use `--scope` then we will need to insert some other intermediate child process to avoid zombies.